File: src/Plotly.NET/Traces/ObjectAbstractions/FinanceMarker.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices

type FinanceMarker() =
    inherit DynamicObj()

    static member init
        (
            ?FillColor: Color,
            ?LineColor: Color,
            ?LineWidth: float,
            ?LineDash: StyleParam.DrawingStyle
        ) =
        FinanceMarker()
        |> FinanceMarker.style (
            ?FillColor = FillColor,
            ?LineColor = LineColor,
            ?LineWidth = LineWidth,
            ?LineDash = LineDash

        )

    static member style
        (
            ?FillColor: Color,
            ?LineColor: Color,
            ?LineWidth: float,
            ?LineDash: StyleParam.DrawingStyle
        ) =
        fun (financeMarker: FinanceMarker) ->

            let line =
                financeMarker.TryGetTypedPropertyValue<Line>("line")
                |> Option.defaultValue(Plotly.NET.Line.init())
                |> Line.style (?Color = LineColor, ?Width = LineWidth, ?Dash = LineDash)

            financeMarker
            |> DynObj.withOptionalProperty "fillcolor" FillColor
            |> DynObj.withProperty "line" line

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/FinanceMarker.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.FinanceMarker FinanceMarker
        (

            Optional<Color> MarkerColor = default,
            Optional<Color> LineColor = default,
            Optional<double> LineWidth = default,
            Optional<StyleParam.DrawingStyle> LineDash = default
        ) => 
            Plotly.NET.TraceObjects.FinanceMarker.init(

                MarkerColor: MarkerColor.ToOption(),
                LineColor: LineColor.ToOption(),
                LineWidth: LineWidth.ToOption(),
                LineDash: LineDash.ToOption()
            );
}

